File: 2432.处理用时最长的那个任务的员工.js
/*
 * @lc app=leetcode.cn id=2432 lang=javascript
 *
 * [2432] 处理用时最长的那个任务的员工
 */

// @lc code=start
/**
 * @param {number} n
 * @param {number[][]} logs
 * @return {number}
 */
var hardestWorker = function(n, logs) {
    let maxTime = -1;
    let employeeId = -1;
    let prevTime = 0;

    for (let i = 0; i < logs.length; i++) {
        const [id, endTime] = logs[i];
        const taskTime = endTime - prevTime;

        if (taskTime > maxTime || (taskTime === maxTime && id < employeeId)) {
            maxTime = taskTime;
            employeeId = id;
        }

        prevTime = endTime;
    }

    return employeeId;
};
// @lc code=end

File: 2432.处理用时最长的那个任务的员工.java
/*
 * @lc app=leetcode.cn id=2432 lang=java
 *
 * [2432] 处理用时最长的那个任务的员工
 */

// @lc code=start
class Solution {
    public int hardestWorker(int n, int[][] logs) {
        int maxTime = logs[0][1];
        int employeeId = logs[0][0];

        for (int i = 1; i < logs.length; i++) {
            int currentTime = logs[i][1] - logs[i - 1][1];
            if (currentTime > maxTime) {
                maxTime = currentTime;
                employeeId = logs[i][0];
            } else if (currentTime == maxTime) {
                employeeId = Math.min(employeeId, logs[i][0]);
            }
        }

        return employeeId;
    }
}
// @lc code=end

